### PR TITLE
Goals & Progress | Parse Overall Progress from PnP

### PR DESCRIPTION
**Pull request by daniel-leod**
#2064

### CODE REVIEW

**Review by CarsonF — CHANGES_REQUESTED** [review]
I was just made aware today that this extracts formatted numbers instead of the actual number values. Several decimal places are lost which are important. 

We can access the actual value much easier than what is done here. 
```ts
sheet.AL20
```
will give us cell `AL20`. There's a `v` property that is the raw number which has all of the precision and doesn't require us to parse it with "parsePercent".

Additionally all the data we require is on a single dynamic row based on fiscal year.
Once we find that row we can grab the data based on the static columns.

Something like this could work.
```ts
const findFiscalYearRow = (sheet, fiscalYear: number) => {
  let i = 20;
  while (true) {
    const cell = sheet[`AG${i}`];
    if (typeof cell.v !== 'number') {
      return null;
    }
    if (cell.v === fiscalYear) {
      return i;
    }
  }
});
```

After that it's simply
```ts
sheet[`AL${fiscalYearRow}`].v
```
For all the columns we need.

This doesn't require `sheet_to_json` either.
We could still have null checks on `v` for sanity though.

---

Given how close these 3 data points actually are I'm totally fine if you want to squash them into one method. If not, then definitely extracting the common pieces into separate methods.

File: src/components/progress-summary/progress-extractor.service.ts [hunk 1]
```diff
@@ -65,15 +58,103 @@ export class ProgressExtractor {
...
+          } else if (parseInt(row.AI) === fiscalQuarter) {
+            return this.parseRawData(row.AI, row.AI);
+          } else if (parseInt(row.AJ) === fiscalQuarter) {
+            return this.parseRawData(row.AJ, row.AJ);
+          } else if (parseInt(row.AK) === fiscalQuarter) {
+            return this.parseRawData(row.AK, row.AK);
+          }
+        }
+      }
+    } catch (e) {
```
Comment (CarsonF) [inline src/components/progress-summary/progress-extractor.service.ts:95]: @blakekrammes what is this catch for?

File: src/components/progress-summary/progress-extractor.service.ts [hunk 2]
```diff
@@ -12,7 +12,11 @@ export class ProgressExtractor {
...
   ) {}
 
-  extractCumulative(pnp: WorkBook, file: FileVersion): Progress | null {
+  extractCumulative(
+    pnp: WorkBook,
+    file: FileVersion,
+    fiscalYear: number
+  ): Progress | null {
     // new standard 2020 version has new sheet "Harvest" which isolates relevant progress data
     const sheet = pnp.Sheets.Harvest ?? pnp.Sheets.Progress;
```
Comment (CarsonF) [inline src/components/progress-summary/progress-extractor.service.ts:21]: We shouldn't need to worry about the Harvest tab either. It doesn't have the data we are looking for.

File: src/components/progress-summary/progress-extractor.service.ts (hunk 1 above)
Comment (blakekrammes) [reply src/components/progress-summary/progress-extractor.service.ts:95]: `parsePercent`, which is called by `parseRawData`, can throw an error.

File: src/components/progress-summary/progress-extractor.service.ts (hunk 2 above)
Comment (blakekrammes) [reply src/components/progress-summary/progress-extractor.service.ts:21]: It doesn't? Isn't `cumulative` the same as `summary` was before?

Comment (blakekrammes) [reply src/components/progress-summary/progress-extractor.service.ts:21]: Oh right, because you're wanting to grab the longer decimal values now.

**Review by CarsonF — APPROVED** [review]

**Pull request merged**